coral/src/domain/topic/topic-api.ts

```diff
@@ -125,7 +125,32 @@ const getTopicRequestsForApprover = (
     .get<KlawApiResponse<"getTopicRequestsForApprover">>(
       `/getTopicRequestsForApprover?${new URLSearchParams(filteredParams)}`
     )
-    .then(transformGetTopicRequestsForApproverResponse);
+    .then(transformGetTopicRequestsResponse);
+};
+
+const getTopicRequests = (
+  params: KlawApiRequestQueryParameters<"getTopicRequests">
+): Promise<TopicRequestApiResponse> => {
+  const filteredParams = omitBy(
+    { ...params, isMyRequest: String(Boolean(params.isMyRequest)) },
+    (value, property) => {
+      const omitIsMyRequest = property === "isMyRequest" && value === "false";
+      const omitSearch = property === "search" && value === "";
+      const omitEnv =
+        property === "env" && (value === "ALL" || value === undefined);
+      const omitOperationType =
+        property === "operationType" &&
+        (value === "ALL" || value === undefined);
+
+      return omitIsMyRequest || omitSearch || omitEnv || omitOperationType;
+    }
+  );
+
+  return api
+    .get<KlawApiResponse<"getTopicRequests">>(
+      `/getTopicRequests?${new URLSearchParams(filteredParams)}`
+    )
+    .then(transformGetTopicRequestsResponse);
 };
```

Comment:
For the falsy values - instead of explicitly doing this for each key, would it make sense to add a filter function that omits the falsy values?

For the ALL values - IMO i would keep it in the url even if it means the same as "no filter". Reason being that it is more explicit and clear if i click on the link from somewhere else on what i get and frankly why not. It's less code to NOT remove it. I do understand that this might be a design choice though, so feel free to ignore if already decided.

Reply:
Yes, this sounds like a good idea! We probably could have this as a util to use in all the relevant spots in `domain`.

Reply:
Unfortunately looks trickier in our case, mostly because of the case when the param value is `"ALL":
- for some params it is a valid value that we want to pass (like `requestStatus`)
- for others it is one we explicitly never want to pass because it will return no result (`environment`)

I think your approach is still good, and should probably be implemented, but I would say it would be a followup housekeeping issue like

```
refactor(coral): Add getParams util to parse URL search params in `domain` API functions
```

Reply:
Sounds good :) I can create the follow up issue.